common/src/main/scala/explore/model/AppContext.scala

```diff
@@ -151,6 +154,31 @@ case class AppContext[F[_]](
   ): Callback = replacePage(appTab, programId, obsId.map(o => ObsIdSet.one(o)), targetId)
 }
 
+object WebWorkers {
+  import scala.scalajs.js
+  import scala.scalajs.js.annotation.JSImport
+
+  /**
+   * This deserves an explanation:
+   *
+   * To make the webworker act correctly in both dev and production we shoud import it as a module
+   * rather than just doing a direct consructor call.
+   *
+   * Doing the import with the "worker" param gives a constructor for the worker which we can wrap
+   * inline lets us save some space keeping a single chunk More info see:
+   * https://vitejs.dev/guide/features.html#import-with-query-suffixes=
+   */
+  @js.native
+  @JSImport("/workers.js?worker&inline", JSImport.Default)
```

Comment:
This is vite specific then? Why `inline`?

Reply:
It is vite specific. `inline` means it wont get built into a separate chunk. The extra chunk in my local tests was fairly big so this seems a good option which also will work well in terser. We can further test this in the future